src/net_processing.cpp

```diff
@@ -1741,7 +1741,10 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
                 // See https://github.com/bitcoin/bitcoin/issues/8279 for details.
                 assert(recentRejects);
                 recentRejects->insert(tx.GetHash());
-            }
+                if (RecursiveDynamicUsage(*ptx) < 100000)
```

Comment:
I was trying to avoid any complication here, but if you think something simple is worth it I'm ok with adding something.